### ISSUE TITLE
Support alembic for the library of custom types

### ISSUE THREAD

**Issue by mehcode** [kvesteri/sqlalchemy-utils#106]
> https://groups.google.com/d/msg/sqlalchemy-alembic/oBAY2K1Xkeo/a0kMlVPdfuwJ
> ---
> 
> Alembic gives you _two_ options to determine how a type object is rendered when using autogenerate.  The first is to implement **repr**() on your type, however that won’t give you access to the parent module.   The second is to implement render_item(), which gives you complete control over how any specific type is rendered.  Implementing this will give you a way to fully control for any specific type how it is rendered.  this is configurable via render_item: http://alembic.readthedocs.org/en/latest/api.html#alembic.environment.EnvironmentContext.configure.params.render_item

---

It looks like we either need to proxy `repr` to the underlying `impl` or implement `render_item` (the former seems easier).

**Comment by bx2** [comment]
Both solutions would not work, because of the [recent bug in Alembic](https://github.com/zzzeek/alembic/pull/17), which is now fixed.

Of course there is still the problem of representation. If you auto-generate from Alembic, then your fields are rendered with types from sqlalchemy-utils, but with errors i.e. UUIDType has no values passed beside the length, which causes an error. Same goes for ChoiceType etc.

**Comment by mehcode** [comment]
Nah.. the idea is to generate with the underlying `impl`. Type decorators aren't meant to go in alembic (from what I understand). 

So a `UUIDType` would ouput a `BINARY` or `CHAR` depending on the dialect and options.

I've done a quick sketch of this and it works, I just haven't had time to do it in full yet.

**Comment by frol** [comment]
Isn't this issue annoying absolutely everybody?!

**Comment by dbrumley** [comment]
Yes, it is annoying me.  One specific case is for `PasswordType`.  Any objection to a PR that fixed this one; it's simple with:
```
   def __repr__(self):
        return "PasswordType(max_length=%r)"%self._max_length
```

**Comment by hsoto92** [comment]
Still having trouble with `UUIDType`. 

```
File "C:\www\liu-passport-api\flask_application\migrations\versions\88f9317a1de1_.py", line 151, in upgrade
    sa.Column('user_id', UUIDType(length=16), nullable=True),
TypeError: <flask_script.commands.Command object at 0x041F26D0>: __init__() got an unexpected keyword argument 'length'
```

Any idea how to fix this?

**Comment by eshlox** [comment]
@hsoto92 Edit migration file and change ```sa.Column('user_id', UUIDType(length=16), nullable=True)``` to ```sa.Column('user_id', UUIDType(max_length=16), nullable=True)```

**Comment by mnowotka** [comment]
+1

**Comment by dxd5581** [comment]
### Issues with ChoiceType

I was having **issues with the ChoiceType**, I posted on below link **how I got around it**:

### https://github.com/kvesteri/sqlalchemy-utils/issues/129#issuecomment-1017329251
